File: docs/examples/extern-struct-example-1.js
import { ext_struct, reg_struct } from './extern-struct-example-1.zig';

console.log('Extern:');
console.log(ext_struct.dataView.getInt16(0, true));
console.log(ext_struct.dataView.getBigInt64(8, true));
console.log('Regular (wrong):');
console.log(reg_struct.dataView.getInt16(0, true));
console.log(reg_struct.dataView.getBigInt64(8, true));
console.log('Regular (correct):');
console.log(reg_struct.dataView.getInt16(8, true));
console.log(reg_struct.dataView.getBigInt64(0, true));

File: docs/examples/extern-struct-example-1.zig
pub const ExternStruct = extern struct {
    small_int: i16,
    big_int: i64,
};

pub const RegularStruct = struct {
    small_int: i16,
    big_int: i64,
};

pub const ext_struct: ExternStruct = .{ .small_int = 123, .big_int = 4567890123 };
pub const reg_struct: RegularStruct = .{ .small_int = 123, .big_int = 4567890123 };
